File: src/BL.EF/Validators/ValidationConstants.cs
namespace KisV4.BL.EF.Validators;

public static class ValidationConstants {
    // string lengths
    public const int MaxNameLength = 30;
    public const int MaxNoteLength = 200;
    public const int MaxUnitNameLength = 10;
    public const int MaxDescriptionLength = 200;

    // pagination
    public const int MaxPageSize = 100;

    // costs
    public const decimal MaxAllowedCost = 100_000;
    public const decimal MaxMarginPercent = 300;

    // amounts
    public const decimal MaxTransactionAmount = 100_000;
    public const decimal MaxCompositionAmount = 100_000;

    // layouts
    public const int LayoutWidth = 4;
    public const int LayoutHeight = 4;
}

File: src/Frontend/ViteApp/src/constants/validationConstants.ts
const validationConstants = Object.freeze({
  maxNameLength: 30,
  maxNoteLength: 200,
  maxUnitNameLength: 10,
  maxDescriptionLength: 200,

  maxAllowedCost: 100_000,
  maxMarginPercent: 300,

  maxTransactionAmount: 100_000,
  maxCompositionAmount: 100_000,

  numberRegex: /^-?(?:0|[1-9]\d*)(?:\.\d+)?$/,
});

export default validationConstants;
